wfcGen.py

```diff
@@ -0,0 +1,264 @@
+import random
+
+from PIL import Image
+
+
+res = 3
+cropShift = False
+cells = {}
+cropped = []
+for ins in range(4):
+    field = Image.open("image.png")
+    if cropShift:
+
+        field = field.rotate(90*ins)
+
+        SX = field.size[0]
+        SY = field.size[1]
+
+        copy = Image.new("RGB", (SX*2, SY*2))
+        copy.paste(field, (0, 0))
+        copy.paste(field, (SX, 0))
+        copy.paste(field, (0, SY))
+        copy.paste(field, (SX, SY))
+        field = copy

```

Comment:
Why?
As python objects are reference types, `field` and `copy` become the same image. Not copies of the same image (which would happen if you did `field = copy.copy()`) but the exact same image. When you modify an object through `field`, the object visible through `copy` is modified the same way (because they point at the same object).

You, most likely, either mean to copy the image (via the `.copy()` method) or can just use one name for the image

Reply:
Then why do you have it in the first place?

It is not very safe to assume you will never use it again. When you come back to this code 1 year later, will you still remember that these two variables are pointing at the same object and you're not supposed to use copy?

You could make your language technically ensure that you don't (e.g., in Python you could write `del copy`) but in this case - why would you? I think it makes more sense to just create the image and call it `field`